src/game.py

```diff
@@ -62,27 +67,29 @@ def handle_winning_game(self):
         self.ui.display_message(self.message.declare_winner(winner))
         self.repeat_game()
 
+    def handle_draw(self):
+        self.ui.display_message(self.message.game_over_message())
+        self.repeat_game()
+
     def handle_full_board(self):
         if self.rules.is_winner(self.game_board):
             self.handle_winning_game()
         else:
-            self.get_prompt(self.total_marks_on_board)
-            self.repeat_game()
+            self.handle_draw()
 
     def take_turns(self):
-        self.get_prompt(self.total_marks_on_board)
-        self.process_user_input()
+        self.prompt_for_move(self.total_marks_on_board)
+        self.handle_mark_board()
```

Comment:
`self.handle_mark_board()` [returns the `self.game_board`](https://github.com/t-keazirian/tic-tac-toe/blob/b7fb0b0a3980f75c9036240bb0dacd5f1b2c2841/src/game.py#L51) but its never captured or used.

Reply:
this has not been addressed yet.  I see what's going on though.  `self.handle_mark_board()` is returning the `self.game_board`, and its only being using for your tests. Typically you do not want to have code that only supports tests.

Reply:
Ah I see.  so can the tests for this be removed then? Because I have coverage from the `mark_board` method in the Board class?  I see that if I remove the return statement in `handle_mark_board`, my tests fail because it is no longer returning anything - but I don't know that we need it to return anything since the board is taking care of that?

Reply:
what is this method doing? is it just extracting game loop code into a helper method?  If so, its basically a private method and probably does not need tested.  Are you familiar with code coverage tools? you can removed those tests, and run code coverage and see wether or not the code is being called via tests.

Reply:
For now; go ahead and leave it.  This PR represents a significant effort to remove some tech debt, but its not here to remove it all.  Make a note, or a trello card, and let's move on.

Reply:
I'm not familiar but I will look into it and create a card.